builder/dockerfile/shell_parser.go

```diff
@@ -169,25 +170,23 @@ func (sw *shellWord) processSingleQuote() (string, error) {
 	//   all the characters (except single quotes, making it impossible to put
 	//   single-quotes in a single-quoted string).
 
-	var result string
+	var result bytes.Buffer
 
 	sw.scanner.Next()
 
+loop:
```

Comment:
I know its a personal preference thing, but I'm really not a fan of this type of `break` - it just feels too much like a goto to me.